recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,155 @@
+                self._libressl_find_pattern,
+                self._libressl_replacement_pattern.format(
+                    LIBRESSL_INCLUDE_DIRS=self.deps_cpp_info["libressl"].include_paths[0],
+                    LIBRESSL_LIBRARY_DIRS=self.deps_cpp_info["libressl"].lib_paths[0]
+                ))
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.options.shared:
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.ssl_provider == 'OPENSSL':
+            self.requires("openssl/1.1.1g")
+        if self.options.ssl_provider == 'LIBRESSL':
+            self.output.warn("Can be broken. Prefer OpenSSL instead of LIBRESSL")
+            self.requires("libressl/3.2.0")
+        if self.options.zlib == 'SYSTEM':
+            self.requires("zlib/1.2.11")
+            if tools.os_info.is_windows:
+                self.output.warn("Usage zlib provided by conan on Windows can be broken")
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD_PARTY_NOTICES", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["bson-1.0", "mongoc-1.0"] if self.options.shared else \
+            ["bson-static-1.0", "mongoc-static-1.0"]
+        self.cpp_info.includedirs.append(os.path.join("include", "libmongoc-1.0"))
+        self.cpp_info.includedirs.append(os.path.join("include", "libbson-1.0"))
+
+    _libressl_find_pattern = '''
+   if (ENABLE_SSL STREQUAL LIBRESSL)
+      include (FindPkgConfig)
+      message ("-- Searching for LibreSSL/libtls")
+      pkg_check_modules (LIBRESSL libtls)
+      if (LIBRESSL_FOUND)
+         message ("--   Found ${LIBRESSL_LIBRARIES}")
+         set (SSL_LIBRARIES ${LIBRESSL_LIBRARIES})
+         if (LIBRESSL_INCLUDE_DIRS)
+           include_directories ("${LIBRESSL_INCLUDE_DIRS}")
+         endif ()
+         link_directories ("${LIBRESSL_LIBRARY_DIRS}")
+         set (LIBRESSL 1)
+      else ()
+         message ("--   Not found")
+      endif ()
+   endif ()
+'''
+
+    _libressl_replacement_pattern = '''
+   if (ENABLE_SSL STREQUAL LIBRESSL)
+        message ("--   USE LIBRESSL include: {LIBRESSL_INCLUDE_DIRS}!")
+        message ("--   USE LIBRESSL library: {LIBRESSL_LIBRARY_DIRS}!")
+        include_directories ("{LIBRESSL_INCLUDE_DIRS}")
```

Comment:
idk, why cmake can't find paths passed by ```conan```

Reply:
It's usually because of the limitation in the CMake generators where the variable casing is the no all caps (but rather uses the casing of the filename)

Reply:
How can i fix it? I think it is a similar problem with conan-zlib on windows.

Reply:
addressed in n-bes#1 + requires #2933 to be merged